lib/SearchAndSort/tests/advancedSearchQueryToRows-test.js

```diff
@@ -39,15 +39,41 @@ describe('advancedSearchQueryToRows', () => {
     });
   });
 
-  describe('when query without match and search option', () => {
-    it('should use default value for match and search option', () => {
-      const query = 'test';
+  describe('when query contains boolean option as part of value', () => {
+    it('should parse query correctly', () => {
+      const query = 'keyword containsAny some text and some other text';
+
+      expect(advancedSearchQueryToRows(query)).to.deep.include({
+        query: 'some text and some other text',
+        bool: '',
+        searchOption: 'keyword',
+        match: 'containsAny',
+      });
+    });
+  });
+
+  describe('when query contains boolean option and match option as part of value, but no search option', () => {
+    it('should parse query correctly', () => {
+      const query = 'keyword containsAny some text and containsAny';
 
       expect(advancedSearchQueryToRows(query)).to.deep.include({
-        query,
+        query: 'some text and containsAny',
         bool: '',
         searchOption: 'keyword',
-        match: 'containsAll',
```

Comment:
I think that when a user selects the `Advanced search` search option, then enters any value and then opens the Advances Search modal, there should be the user input in the modal's query field.

Reply:
Do you mean like this?

https://github.com/folio-org/stripes-smart-components/assets/19309423/190c443b-beba-446c-b9eb-220c123efb47